addons/io/org.openhab.io.transport.modbus/src/main/java/org/openhab/io/transport/modbus/ModbusBitUtilities.java

```diff
@@ -148,6 +149,48 @@ public static DecimalType extractStateFromRegisters(ModbusRegisterArray register
         }
     }
 
+    /**
+     * Read data from registers and convert the result to StringType
+     *
+     *
+     * @param registers
+     *            list of registers, each register represent 16bit of data
+     * @param index
+     *            zero based item index. Registers are handles as 16bit registers
+     * @param length
+     *            length of string in 8bit characters
+     *
+     * @return string representation queried value
+     * @throws IllegalArgumentException when <tt>index</tt> is out of bounds of registers
+     *
+     */
+    public static StringType extractStringFromRegisters(ModbusRegisterArray registers, int index, int length) {
+        if (index * 2 + length > registers.size() * 2) {
+            throw new IllegalArgumentException(
+                    String.format("Index=%d with length=%d is out-of-bounds given registers of size %d", index, length,
+                            registers.size()));
+        }
+        if (index < 0) {
+            throw new IllegalArgumentException("Negative index are values not supported");
+        }
+        if (length < 0) {
+            throw new IllegalArgumentException("Negative string lenght is not supported");
+        }
+        StringBuffer buff = new StringBuffer(length);
+
+        int src = index;
+        for (int dest = 0; dest < length; dest++) {
+
+            if (dest % 2 == 0) {
+                buff.append((char) ((registers.getRegister(src).getValue() >> 8)));
```

Comment:
How about using  [`String(byte[], Charset)`](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#String(byte[],%20java.nio.charset.Charset)) and let the caller define the charset? 

I am thinking if this would make sense this is the transport module and should be fairly general. 

Furthermore, I think the Register class has a method for accessing bytes directly without masking, but either way is fine I think.

Finally, we can look if RegisterArrays would be able to expose all data as byte[]... would make this implementation more simple/straightforward?

Reply:
You're right with the need of the Charset. I've added this and updated the tests.

An exposed byte[] from RegisterArrays could ease some of the work here, but anyway it would be necessary to cut out a part of the array.

Reply:
Looks good.